keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -129,12 +145,101 @@ def __init__(self):
     def auto_vectorize(self, auto_vectorize):
         self._auto_vectorize = auto_vectorize
 
-    @property
-    def _map_fn(self):
+    def compute_image_signature(self, images):
+        """Computes the output image signature for the `augment_image()` function.
+
+        Must be overridden to return tensors with different shapes than the input
+        images.  By default returns either a `tf.RaggedTensorSpec` matching the input
+        image spec, or a `tf.TensorSpec` matching the input image spec.
+        """
+        if self.force_output_dense_images:
+            return tf.TensorSpec(images.shape[1:], self.compute_dtype)
+        if self.force_output_ragged_images or isinstance(images, tf.RaggedTensor):
+            ragged_spec = tf.RaggedTensorSpec(
+                shape=images.shape[1:],
+                ragged_rank=1,
+                dtype=self.compute_dtype,
+            )
+            return ragged_spec
+        return tf.TensorSpec(images.shape[1:], self.compute_dtype)
+
+    # TODO(lukewood): promote to user facing API if needed
+    def _compute_bounding_box_signature(self, bounding_boxes):
```

Comment:
any reason why this is private method, but `compute_image_signature` is public method?

Reply:
Right now I see no use case to customize the bounding box output signature